pkg/operator/ceph/controller/predicate.go

```diff
@@ -84,6 +84,9 @@ func WatchControllerPredicate() predicate.Funcs {
 				} else if objectToBeDeleted(objOld, objNew) {
 					logger.Debugf("CR %q is going be deleted", objNew.Name)
 					return true
+				} else if objOld.Status.ObservedGeneration != objNew.GetGeneration() {
```

Comment:
`objOld.Status` could be nil. But perhaps this code is not needed anyway, based on the other comment.

Reply:
A number of CI actions are failing, I suspect this fix is needed, though I didn't dig into the failures.
```suggestion
				} else if objOld.Status != nil && objOld.Status.ObservedGeneration != objNew.GetGeneration() {
```

Reply:
✔️ although IMO it's unrelated